server/mlabns/tests/test_resolver.py

```diff
@@ -1,490 +1,608 @@
-        mock_query = mock.Mock(latitude=None, longitude=None)
-        results = geo_resolver.answer_query(mock_query)
-        self.assertEqual(1, len(results))
-        self.assertEqual('valid_candidate', results[0])
-
-    def testAnswerQueryAllCandidatesSameSite(self):
-
-        class GeoResolverMockup(resolver.GeoResolver):
-            def get_candidates(self, unused_arg):
-                mock_tool = mock.Mock(site_id='valid_site_id', latitude=2.0,
-                                      longitude=1.0)
-                return [mock_tool, mock_tool]
-
-        geo_resolver = GeoResolverMockup()
-        mock_query = mock.Mock(latitude=0.0, longitude=0.0)
-        results = geo_resolver.answer_query(mock_query)
-        self.assertEqual(1, len(results))
-        self.assertEqual(2.0, results[0].latitude)
-        self.assertEqual(1.0, results[0].longitude)
-
-    def testAnswerQueryAllCandidatesDifferentSitesOneClosest(self):
-
-        class GeoResolverMockup(resolver.GeoResolver):
-            def get_candidates(self, unused_arg):
-                return [mock.Mock(site_id='a', latitude=2.0, longitude=1.0),
-                        mock.Mock(site_id='b', latitude=20.0, longitude=34.9)]
-
-        geo_resolver = GeoResolverMockup()
-        mock_query = mock.Mock(latitude=0.0, longitude=0.0)
-        results = geo_resolver.answer_query(mock_query)
-        self.assertEqual(1, len(results))
-        self.assertEqual(2.0, results[0].latitude)
-        self.assertEqual(1.0, results[0].longitude)
-
-    def testAnswerQueryAllCandidatesDifferentSitesMultipleClosest(self):
-
-        class GeoResolverMockup(resolver.GeoResolver):
-            def get_candidates(self, unused_arg):
-                return [mock.Mock(site_id='a', latitude=2.0, longitude=1.0),
-                        mock.Mock(site_id='b', latitude=20.0,longitude=34.9),
-                        mock.Mock(site_id='c', latitude=2.0, longitude=1.0)]
-
-        geo_resolver = GeoResolverMockup()
-        mock_query = mock.Mock(latitude=0.0, longitude=0.0)
-        results = geo_resolver.answer_query(mock_query)
-        self.assertEqual(1, len(results))
-        self.assertEqual(2.0, results[0].latitude)
-        self.assertEqual(1.0, results[0].longitude)
+        tool_fetcher_patch = mock.patch.object(
+            tool_fetcher, 'ToolFetcher', autospec=True)
+        self.addCleanup(tool_fetcher_patch.stop)
+        tool_fetcher_patch.start()
+        self.resolver = resolver.GeoResolver()
+
+    def testAnswerQueryWhenSingleToolIsClosest(self):
+        """When a single tool is closest, return that tool."""
+        query = lookup_query.LookupQuery()
+        query.tool_id = _TOOL_ID
+        query.latitude = 0.0
+        query.longitude = 0.0
+
+        close_tool = _createSliverTool(
+            _TOOL_ID, site_id='abc01', latitude=1.0, longitude=1.0)
+        far_tool = _createSliverTool(
+            _TOOL_ID, site_id='cba01', latitude=5.0, longitude=5.0)
```

Comment:
I think that's out of scope for this pull request as that's the job of server/mlabns/util/distance.py and I'm not touching that code. But it is a good test case and something that's currently not exercised in test_distance.py. Can you file a bug / submit a separate pull request?